source/client/flush_worker_impl.h

```diff
@@ -0,0 +1,51 @@
+// Flush worker implementation. Flush worker periodically flushes metrics
+// snapshot to all configured stats sinks in Nighthawk.
+#pragma once
+
+#include <vector>
+
+#include "envoy/api/api.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/stats/sink.h"
+#include "envoy/stats/store.h"
+#include "envoy/thread_local/thread_local.h"
+
+#include "common/worker_impl.h"
+
+namespace Nighthawk {
+namespace Client {
+
+// Only a single live flush worker instance can be created in Nighthawk at any
+// time.
+// Flush worker periodically flushes metrics snapshot to all configured stats sinks in Nighthawk. It
+// will keep running until exitDispatcher() gets called after all client workers are completed in
+// process_impl.cc. It will make the last flush before shutdown in shutdownThread().
+class FlushWorkerImpl : public WorkerImpl {
+public:
+  FlushWorkerImpl(Envoy::Api::Api& api, Envoy::ThreadLocal::Instance& tls,
```

Comment:
Can we document the constructor and its arguments?

Reply:
Done